README.md

```diff
@@ -315,7 +315,7 @@ interface Observable {
   constructor(SubscribeCallback callback);
   undefined subscribe(Observer observer);
 
-  undefined finally(VoidFunction callback);
+  Observable finally(VoidFunction callback);
```

Comment:
Also, does `finally()` still need the `VoidFunction` callback here? Probably not, right, if we're observing completion via the returned Observable. That brings me to one more question: why return an Observable instead of a Promise?

If we're just trying to understand if the Observable finished in any manner (regardless of whether it did so in error or successfully), like `Promise.prototype.finally()`, maybe this should just be a Promise-returning API?

At the very least, we need to update https://github.com/domfarolino/observable/blob/master/README.md?plain=1#L427-L429 in this PR.

Reply:
Yeah, it needs it. `finally` callback would fire if the subscription ends for any reason: error, complete, or abort/unsubscribe.

Reply:
Isn't that what we're using the returned Observable for? Why have both? It seems that this should either:
 - Return undefined and take a callback
 - Return an Observable (whose `complete()` callback is called whenever the observable instance that `finally()` was called on ends for any reason) and takes **no** callback

Are both needed?

Reply:
We discussed this offline, and we worked it out. I think we agreed this can be merged, @domfarolino

Reply:
Maybe we could just add some documentation just above this method mentioning that this is a pass-through for the observable that this is called on (i.e., the returned observable is "identical"), but that the `finally()` handler gets called.... *after*(?) the complete()/error() events of the final observable?

Reply:
it would technically get called _after_ the error or `error`, `complete`.

```ts
source
  .finally(() => console.log('finalized'))
  .subscribe({
    complete: () => console.log('complete'),
  });
  
// logs
// "complete"
// "finalized"
```

Reply:
Oh... looks like I already did this work... just circling through all of this.